src/parser.ts

```diff
@@ -60,7 +61,6 @@ export class Parser {
     readonly scanner: Scanner;
     readonly operatorPrecedence: any;
 
-    readonly sourceType: string;
```

Comment:
What's the reason for changing this over to `isModule`? Previously you had argued for a string type to support eventual additional sourceTypes. Is that YAGNI now?